client/pages/conversation.jsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import Redirect from '../components/redirect';
+import AppContext from '../lib/app-context';
+import formatDate from '../lib/format-date';
+
+export default class Conversation extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      messages: null,
+      currentMessage: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSend = this.handleSend.bind(this);
+    this.refresh = this.refresh.bind(this);
+  }
+
+  refresh() {
+    const { otherId } = this.props;
+    const { userId } = this.context.user;
+    fetch(`/api/message/convo/${userId}/${otherId}`)
+      .then(response => response.json())
+      .then(messages => this.setState({ messages }));
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSend(event) {
+    event.preventDefault();
+    if (!this.state.currentMessage) {
+      return null;
+    }
+    const { currentMessage } = this.state;
+    const { otherId, postId } = this.props;
+    const { username, userId } = this.context.user;
+    const header = new Headers();
+    header.append('Content-Type', 'application/json');
+    let body = {
+      senderId: userId,
+      senderName: username,
+      recipientId: otherId,
+      content: currentMessage,
+      postId
+    };
```

Comment:
Instead of putting the userId in the body or the URL path, put the token in your headers. the way you have it set up now, anybody can grab any messages for any user.